### ISSUE TITLE
iOS 14 compatibility issues

### ISSUE THREAD

**Issue by nylki** [alexdrone/Store#28]
Hi!
I was trying to get the getting-started code snippets working in a freshly created swiftUI project that targets iOS 14.
But I am running into issues.

First, perhaps unrelated, I had to bump the iOS version in `Package.swift` from v12 to v13 in a fork, to able to resolve it inside XCode as a SwiftPackage, because of features like `Future` requiring at least iOS v13.

After getting the package installed I tried out the snippet in my ContentView, but got stuck at following errors:
![Screenshot 2021-02-03 at 12 19 40](https://user-images.githubusercontent.com/1710598/106739981-23c18500-661a-11eb-9eb2-4457071ebd38.png)

**Comment by alexdrone** [comment]
Have you tried by having the action type explicit - as in CounterAction.increase(amount: 1) ?

**Comment by nylki** [comment]
Thank you for the quick response alex! I followed your suggestion and tried it. It resolved those errors but created new ones:
![Screenshot 2021-02-04 at 17 39 30](https://user-images.githubusercontent.com/1710598/106925386-55b51300-6710-11eb-843c-cc19d9533672.png)

For reference, here is my full ContentView.swift:

```.swift
//  ContentView.swift

import SwiftUI
import Store

struct Counter { var count = 0 }

enum CounterAction: Action {
  case increase(amount: Int)
  case decrease(amount: Int)

  func reduce(context: TransactionContext<Store<Counter>, Self>) {
    defer {
      context.fulfill()
    }
    switch self {
    case .increase(let amount):
      context.reduceModel { $0.count += amount }
    case .decrease(let amount):
      context.reduceModel { $0.count -= amount }
    }
  }

  func cancel(context: TransactionContext<Store<Counter>, Self>) { }
}


// MARK: - UI
struct ContentView: View {
    @StateObject var store = Store<Counter>(model: Counter())
    
    var body: some View {
//        Text("Text without Logic")
        Text("counter \(store.model.count)").onTapGesture {
            store.run(action: CounterAction.increase(amount: 1))
        }
    }
}

struct ContentView_Previews: PreviewProvider {
    static var previews: some View {
        ContentView().background(Color.gray)
    }
}

```

**Comment by alexdrone** [comment]
Hi! sorry for the late reply :(

This is now fixed in Store 3.6

**Comment by nylki** [comment]
@alexdrone No worries, thanks!
